src/web/components/SharingPermission/ParticipantsTable.tsx

```diff
@@ -9,20 +9,34 @@ import { ParticipantItem } from './ParticipantItem';
 
 import './ParticipantsTable.scss';
 
-type ParticipantsTableProps = {
-  participants: AvailableParticipantDTO[];
+export type SharingParticipant = AvailableParticipantDTO & {
+  addedBy: 'Manual' | 'Auto' | 'Manual / Auto';
+};
+
+type ParticipantsTableBaseProps<T> = {
+  participants: T[];
   filterText: string;
   selectedTypeIds?: Set<number>;
   onSelectedChange: (selectedItems: Set<number>) => void;
   selectedParticipantIds?: Set<number>;
-  tableHeader: (filteredParticipants: AvailableParticipantDTO[]) => ReactNode;
+  tableHeader: (filteredParticipants: T[]) => ReactNode;
   className?: string;
   hideSelectAllCheckbox?: boolean;
   showAddedByColumn?: boolean;
-  onFilteredParticipantChanged?: (filteredParticipants: AvailableParticipantDTO[]) => void;
+  onFilteredParticipantChanged?: (filteredParticipants: T[]) => void;
+};
+
+type DetermineParticipantsType<ShowColumn extends boolean> = ShowColumn extends true
```

Comment:
I can see how the name of this type could be confusing, since participants themselves have a `type`, whereas this is referring to a typescript type. Not sure if something like `DetermineParticipantTSType` is any better...